### PR TITLE
Add Tailwind to advance options.

### LINKED ISSUE #186: [Feature Request] Adding Tailwind Configuration to the Initial Setup
### Tell us about your feature request

Right now, we have everything ready for web development, but Tailwind is part of my stack, and I need to configure it manually. So, adding a Tailwind option for the initial setup should be fine.

### Disclaimer

- [X] I agree

### PR DESCRIPTION
**Pull request by PrajvalBadiger**
By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

## Problem/Feature
Resolves #186 

## Description of Changes: 
This PR adds an option to the `--advanced` flag that let's users add Tailwind CSS framework to the project
- This change creates package.json, tailwind.config.js, input.css and output.css files
- pnpm is used to install tailwind via package.json as I couldn't test with npm
- updated Makefile to run tailwind for the build recipe. which is also used in .air.toml

## Checklist

- [x] I have self-reviewed the changes being requested
- [x] I have updated the documentation (if applicable)

### CODE REVIEW

**Comment by briancbarrow** [comment]
@PrajvalBadiger Thanks for submitting this PR. I wanted to ask what you thought about changing the implementation to use the [Tailwind Standalone CLI](https://tailwindcss.com/blog/standalone-cli) instead of installing the package with Node.js? 

I've used the standalone CLI in my projects and it works well. We'd have to add some language to the output saying to make sure they have it installed, but we are already doing something similar with Templ so I don't think that will be an issue. 

The other thing we probably want to add is some checks to make sure that the tailwind option is not used without the htmx option.

**Comment by PrajvalBadiger** [comment]
> @PrajvalBadiger Thanks for submitting this PR. I wanted to ask what you thought about changing the implementation to use the [Tailwind Standalone CLI](https://tailwindcss.com/blog/standalone-cli) instead of installing the package with Node.js?
> 
> I've used the standalone CLI in my projects and it works well. We'd have to add some language to the output saying to make sure they have it installed, but we are already doing something similar with Templ so I don't think that will be an issue.
> 

@briancbarrow, Thanks for the review. I was not aware of the Tailwind Standalone CLI and had used Node.js for Tailwind in my projects. A huge dependency on Node.js would be reduced by using the standalone CLI.

> The other thing we probably want to add is some checks to make sure that the tailwind option is not used without the htmx option.

There a few ways we can implement this:
1. By Default selecting the tailwind option is disabled and user can only select tailwind option if htmx option is selected
2. If tailwind option is selected, htmx is automatically selected
3. If only the tailwind option is used without htmx option, the program fails and print an error. If this is the case the user has to redo all the steps and select htmx option with tailwind option.

**Comment by briancbarrow** [comment]
> There a few ways we can implement this:

> By Default selecting the tailwind option is disabled and user can only select tailwind option if htmx option is selected
> If tailwind option is selected, htmx is automatically selected
> If only the tailwind option is used without htmx option, the program fails and print an error. If this is the case the user has to redo all the steps and select htmx option with tailwind option.

I would be good with either the first or second option.

**Comment by PrajvalBadiger** [comment]
1. I have implemented the second option - If tailwind option is selected, htmx is automatically selected.
2. Node.js dependency is droped and tailwind standalone cli can be used now.
3. Added note to remind users to install tailwind standalone cli.

File: cmd/create.go [hunk 1]
```diff
@@ -244,6 +244,12 @@ var createCmd = &cobra.Command{
 		fmt.Println(endingMsgStyle.Render("\nNext steps:"))
 		fmt.Println(endingMsgStyle.Render(fmt.Sprintf("• cd into the newly created project with: `cd %s`\n", project.ProjectName)))
 
+		if options.Advanced.Choices["Tailwind"] {
+			options.Advanced.Choices["Htmx"] = true
+			fmt.Println(endingMsgStyle.Render("• Install the tailwind standalone cli if you haven't already, grab the executable for your platform from the latest release on GitHub\n"))
+			fmt.Println(endingMsgStyle.Render("• Tailwind GitHub: https://github.com/tailwindlabs/tailwindcss/releases/\n"))
```
Comment (briancbarrow) [inline cmd/create.go:250]: I would change this to point to the page explaining the Standalone CLI. That page will give more info about it and how to install it. It is still a relatively unknown feature so pointing people to an explainer page will be helpful. 
```suggestion
			fmt.Println(endingMsgStyle.Render("• More info about the Tailwind CLI: https://tailwindcss.com/blog/standalone-cli\n"))
```

File: cmd/steps/steps.go [hunk 2]
```diff
@@ -108,6 +108,11 @@ func InitSteps(projectType flags.Framework, databaseType flags.Database) *Steps
 						Title: "Websocket endpoint",
 						Desc:  "Add a websocket endpoint",
 					},
+					{
+						Flag:  "Tailwind",
+						Title: "TailwindCSS",
+						Desc:  "A utility-first CSS framework",
```
Comment (briancbarrow) [inline cmd/steps/steps.go:114]:
```suggestion
						Desc:  "A utility-first CSS framework (selecting this will automatically add HTMX)",
```

**Review by briancbarrow — CHANGES_REQUESTED** [review]
This is looking great. Some small changes to make things clearer in the output of the cli.

**Comment by briancbarrow** [comment]
Thanks @PrajvalBadiger! The only thing left is it looks like there are commits on here that are from previously merged PRs. Can you do a rebase and only keep the commits that are relevant to this PR?

File: README.md [hunk 3]
```diff
@@ -111,6 +111,9 @@ You can now use the `--advanced` flag when running the `create` command to get a
 - [HTMX](https://htmx.org/) support using [Templ](https://templ.guide/)
 - CI/CD workflow setup using [Github Actions](https://docs.github.com/en/actions)
 - [Websocket](https://pkg.go.dev/nhooyr.io/websocket) sets up a websocket endpoint
+- [Tailwind](https://tailwindcss.com/) Css framework
+
+Note: selecting tailwind option automatically selects htmx.
```
Comment (Melkeydev) [inline README.md:116]: nice

File: cmd/program/program.go [hunk 4]
```diff
@@ -216,7 +216,7 @@ func (p *Project) CreateMainFile() error {
 	// check if AbsolutePath exists
 	if _, err := os.Stat(p.AbsolutePath); os.IsNotExist(err) {
 		// create directory
-		if err := os.Mkdir(p.AbsolutePath, 0754); err != nil {
+		if err := os.Mkdir(p.AbsolutePath, 0o754); err != nil {
```
Comment (Melkeydev) [inline cmd/program/program.go:219]: What is this doing?

**Review by briancbarrow — APPROVED** [review]
LGTM

File: cmd/program/program.go (hunk 4 above)
Comment (PrajvalBadiger) [reply cmd/program/program.go:219]: This is nothing to do with the PR itself, go lsp in neovim auto formats octals 0o.

**Pull request merged**